File: templates/Uwp/Features/ShareTarget.CodeBehind._VB/TemplateSelectors/SharedContentTemplateSelector.vb
Imports Windows.ApplicationModel.DataTransfer
Imports Param_RootNamespace.Models

Namespace TemplateSelectors
    Public Class SharedContentTemplateSelector
        Inherits DataTemplateSelector

        Public Property DefaultTemplate As DataTemplate

        Public Property StorageItemsTemplate As DataTemplate

        Public Property WebLinkTemplate As DataTemplate

        Public Sub New()
        End Sub

        Protected Overrides Function SelectTemplateCore(item As Object, container As DependencyObject) As DataTemplate
            Dim sharedData = TryCast(item, SharedDataModelBase)
            If sharedData IsNot Nothing Then
                If sharedData.DataFormat = StandardDataFormats.WebLink Then
                    Return WebLinkTemplate
                ElseIf sharedData.DataFormat = StandardDataFormats.StorageItems Then
                    Return StorageItemsTemplate
                End If
            End If

            Return DefaultTemplate
        End Function
    End Class
End Namespace

File: templates/Uwp/Features/ShareTarget.CodeBehind/TemplateSelectors/SharedContentTemplateSelector.cs
using System;
using Windows.ApplicationModel.DataTransfer;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;
using Param_RootNamespace.Models;

namespace Param_RootNamespace.TemplateSelectors
{
    public class SharedContentTemplateSelector : DataTemplateSelector
    {
        public DataTemplate DefaultTemplate { get; set; }

        public DataTemplate StorageItemsTemplate { get; set; }

        public DataTemplate WebLinkTemplate { get; set; }

        public SharedContentTemplateSelector()
        {
        }

        protected override DataTemplate SelectTemplateCore(object item, DependencyObject container)
        {
            var sharedData = item as SharedDataModelBase;
            if (sharedData != null)
            {
                if (sharedData.DataFormat == StandardDataFormats.WebLink)
                {
                    return WebLinkTemplate;
                }
                else if (sharedData.DataFormat == StandardDataFormats.StorageItems)
                {
                    return StorageItemsTemplate;
                }
            }

            return DefaultTemplate;
        }
    }
}
